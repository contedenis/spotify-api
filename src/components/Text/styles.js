// @packages
import styled from 'styled-components';

const TextStyled = styled.h1`
  color: ${({ color }) => color};
  font-size: ${({ size }) => `${size}px`};
  margin: 0;
  ${({ ellipsis }) => ellipsis && `
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis; 
  `};
`;

export {
  TextStyled,
};
