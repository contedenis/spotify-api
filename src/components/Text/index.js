// @packages
import React from 'react';
import PropTypes from 'prop-types';

// @own
import { TextStyled } from './styles';

function Text({
  children,
  color,
  size,
  type,
}) {
  return (
    <TextStyled
      as={type}
      color={color}
      size={size}
    >
      {children}
    </TextStyled>
  );
}

Text.defaultProps = {
  color: 'white',
  size: 12,
  type: 'h3',
};

Text.propTypes = {
  children: PropTypes.node.isRequired,
  color: PropTypes.string,
  size: PropTypes.number,
  type: PropTypes.string,
};

export default Text;
