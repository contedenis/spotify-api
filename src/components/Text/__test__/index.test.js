// @packages
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

// @own
import Text from '../index';

describe('Text', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <Text />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('render text', () => {
    render(
      <Text>
        children
      </Text>,
    );

    const children = screen.getByRole('heading', { name: 'children' });
    expect(children).toBeInTheDocument();
  });
});
