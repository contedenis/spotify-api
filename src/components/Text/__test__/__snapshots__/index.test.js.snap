// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text matches snapshot 1`] = `
<h3
  class="sc-AxjAm bhdljG"
  color="white"
  size="12"
/>
`;
